app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,436 @@
+                convertView, parent);
+            addHandlers(newView);
+            return newView;
+        }
+
+        @Override
+        public View getGroupView(int groupPosition, boolean isExpanded, View convertView,
+                                ViewGroup parent) {
+            View newView = super.getGroupView(groupPosition, isExpanded, convertView, parent);
+            addHandlers(newView);
+            return newView;
+        }
+
+        /**
+         * This method is being called by {@code #getChildView} and {@code #getGroupView} to
+         * customize the list items depending on the message specifications.
+         * It add the click handlers to the buttons or hide them if they aren't being used.
+         * @param newView the inflated view by {@code #getChildView} and {@code #getGroupView}
+         */
+        private void addHandlers(View newView) {
+            TextView idView = (TextView) newView.findViewById(R.id.snackbar_id);
+            final String id = idView.getText().toString();
+            TextView priorityView = (TextView) newView.findViewById(R.id.snackbar_priority);
+            final String priority = priorityView.getText().toString();
+            Message m = mMessagesList.get(new MessageKey(Integer.parseInt(id), Integer.parseInt(priority)));
+
+            // Set action handler
+            TextView actionButton = (TextView) newView.findViewById(R.id.snackbar_action);
+            if ((m.actionHandler != null) && (!m.actionString.isEmpty())) {
+                actionButton.setOnClickListener(m.actionHandler);
+                actionButton.setVisibility(View.VISIBLE);
+            } else {
+                actionButton.setVisibility(View.GONE);
+            }
+
+            // Set Dismiss handler
+            ImageView dismissButton = (ImageView) newView.findViewById(R.id.snackbar_dismiss);
+            if (m.isDismissible) {
+                dismissButton.setOnClickListener(new View.OnClickListener() {
+                    @Override public void onClick(View v) {
+                        dismiss(
+                            new MessageKey(Integer.parseInt(id), Integer.parseInt(priority))
+                        );
+                    }
+                });
+                dismissButton.setVisibility(View.VISIBLE);
+            } else {
+                dismissButton.setVisibility(View.GONE);
+            }
+        }
+    }
+
+    /**
+     * The key of the {@code TreeMap} used to storing the messages.
+     * Helps the {@code TreeMap} maintain it's balance using priority as a parameter.
+     */
+    private class MessageKey implements Comparable<MessageKey> {
+
+        protected int id;
+        protected int priority;
+
+        public MessageKey(int id, int priority){
+            this.id = id;
+            this.priority = priority;
+        }
+
+        @Override public int compareTo(@NonNull MessageKey another) {
```

Comment:
I need to compare IDs because this method is being called by the TreeMap get method.
It will return the key with the same id disregarding the priority.